app/src/main/java/com/android/solvit/seeker/ui/service/Services.kt

```diff
@@ -105,79 +123,123 @@ fun ServicesScreen(
+            Modifier.size(40.dp)
+                .clip(CircleShape)
+                .clickable { navigationActions.navigateTo(Route.PROFILE) }
+                .testTag("servicesScreenProfileImage"))
+        Column(horizontalAlignment = Alignment.Start) {
+          // User's Location
+          Row(verticalAlignment = Alignment.CenterVertically) {
+            Icon(
+                imageVector = Icons.Outlined.LocationOn,
+                contentDescription = "Location Icon",
+                tint = OnPrimary,
+                modifier =
+                    Modifier.size(16.dp).padding(end = 4.dp).testTag("servicesScreenLocationIcon"))
+
+            Text(
+                text =
+                    if (address.length > 27) { // Replace with `userProfile.address`
+                      address.take(27) + "..."
+                    } else {
+                      address
+                    },
+                fontSize = 16.sp,
+                fontWeight = FontWeight.Medium,
+                color = OnPrimary,
+                modifier = Modifier.testTag("servicesScreenCurrentLocation"))
+          }
+
+          // User's Name
+          Text(
+              text = userProfile.name,
+              modifier = Modifier.testTag("servicesScreenUserName"),
+              fontSize = 20.sp,
+              fontWeight = FontWeight.Bold,
+              color = DarkBlue // Dark blue for emphasis
+              )
+        }
+        IconButton(onClick = { toast.show() }, modifier = Modifier.testTag("servicesScreenMenu")) {
+          Icon(imageVector = Icons.Default.Menu, contentDescription = null)
+        }
+      }
+      DockedSearchBar(
+          query = searchText,
+          onQueryChange = searchViewModel::onSearchTextChange,
+          onSearch = searchViewModel::onSearchTextChange,
+          active = isSearching,
+          onActiveChange = { searchViewModel.onToggleSearch() },
+          modifier = Modifier.offset(y = 20.dp).testTag("servicesScreenSearchBar"),
+          placeholder = { Text("Find services near you") },
+          leadingIcon = { Icon(imageVector = Icons.Default.Search, contentDescription = null) },
+          colors = SearchBarDefaults.colors(containerColor = OnPrimary)) {
+            LazyColumn(
+                modifier = Modifier.padding(16.dp),
+                verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                  items(searchResults.size) { index ->
+                    Text(
+                        Services.format(searchResults[index].service),
+                        modifier =
+                            Modifier.clickable {
+                              listProviderViewModel.selectService(searchResults[index].service)
+                              navigationActions.navigateTo(Route.PROVIDERS)
+                            })
+                  }
+                }
+          }
+    }
+  }
+}
+
+@Composable
+fun DiscountSection(
+    navigationActions: NavigationActions,
```

Comment:
I don't know about this, at this stage of the project we cannot add UI elements that are not implemented, and I don't think it is a priority to implement discounts

Reply:
We can change later for a picture in the announcements section rather than discounts